api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -11,88 +12,128 @@
 
-  @Id
-  @GeneratedValue(strategy = GenerationType.IDENTITY)
-  @Column(name = "institution_id")
-  private long dbId; // primary opaque key for DB use only
-
-  @Column(name = "short_name", nullable = false, unique = true)
+  private long institutionId; // primary opaque key for DB use only
   private String shortName; // unique key exposed to API
-
-  @Column(name = "display_name", nullable = false)
   private String displayName;
-
-  @Column(name = "organization_type_enum")
   private Short organizationTypeEnum;
-
-  @Column(name = "organization_type_other_text")
   private String organizationTypeOtherText;
-
-  @OneToMany(mappedBy = "institution", fetch = FetchType.LAZY)
-  private Set<DbInstitutionEmailDomain> emailDomains;
-
-  @OneToMany(mappedBy = "institution", fetch = FetchType.LAZY)
-  private Set<DbInstitutionEmailAddress> emailAddresses;
+  @NotNull private Set<DbInstitutionEmailDomain> emailDomains = Sets.newHashSet();
+  @NotNull private Set<DbInstitutionEmailAddress> emailAddresses = Sets.newHashSet();
 
   public DbInstitution() {}
 
-  public DbInstitution(final String shortName, final String displayName) {
-    setShortName(shortName);
-    setDisplayName(displayName);
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name = "institution_id")
+  public long getInstitutionId() {
+    return institutionId;
   }
 
+  public DbInstitution setInstitutionId(long institutionId) {
+    this.institutionId = institutionId;
+    return this;
+  }
+
+  @Column(name = "short_name", nullable = false, unique = true)
   public String getShortName() {
     return shortName;
   }
 
-  public void setShortName(String shortName) {
+  public DbInstitution setShortName(String shortName) {
     this.shortName = shortName;
+    return this;
   }
 
+  @Column(name = "display_name", nullable = false)
   public String getDisplayName() {
     return displayName;
   }
 
-  public void setDisplayName(String displayName) {
+  public DbInstitution setDisplayName(String displayName) {
     this.displayName = displayName;
+    return this;
   }
 
-  public Short getOrganizationTypeEnum() {
-    return organizationTypeEnum;
+  @Column(name = "organization_type_enum")
+  public OrganizationType getOrganizationTypeEnum() {
+    return DbStorageEnums.organizationTypeFromStorage(organizationTypeEnum);
   }
 
-  public void setOrganizationTypeEnum(Short organizationTypeEnum) {
-    this.organizationTypeEnum = organizationTypeEnum;
+  public DbInstitution setOrganizationTypeEnum(OrganizationType type) {
+    this.organizationTypeEnum = DbStorageEnums.organizationTypeToStorage(type);
+    return this;
   }
 
+  @Column(name = "organization_type_other_text")
   public String getOrganizationTypeOtherText() {
     return organizationTypeOtherText;
   }
 
-  public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
+  public DbInstitution setOrganizationTypeOtherText(String organizationTypeOtherText) {
```

Comment:
So should we assert that the enum value is actually other when setting this? Should we just go ahead and update it if we're setting this? They're a package deal.

Reply:
We have no real requirement for this, so I'm keeping it loose to avoid added complexity.